### DISCUSSION TITLE
converting int into string on where clause

### DISCUSSION THREAD

**Discussion by afonseca69** [simolus3/drift#1418] [Q&A] [answered]
how can I join string field with integer field on where clause?
..where((t) => (t.nome + t.inscricao + t.clienteid).contains(pdescricao))
t.clienteid is integer and the other two are strings

**Comment by simolus3** [comment, marked as answer]
You can cast the integer to a string like this:

```dart
t.nome + t.inscricao + t.clienteid.cast<String>()
```

So the final predicate would look like `(t.nome + t.inscricao + t.clienteid.cast<String>()).contains(pdescricao)`

**Comment by afonseca69** [comment]
thank you


Atenciosamente,


[SoftServ Sistemas]<http://www.softserv.com.br/>[Google+]<https://plus.google.com/+SoftservBr1994> [Curta nossa pÃ¡gina] <https://www.facebook.com/softservsistemas/>  [Conecte-se ao Linkedin] <https://www.linkedin.com/company/softserv-sistemas>  [Siga-nos no Twitter] <https://twitter.com/softserv1994>  [Siga-nos no Instagram] <https://www.instagram.com/softservsistemas>  [Inscreva-se em nosso canal] <http://www.youtube.com/c/SoftservBr1994>    André Fonseca
Pesquisa e Desenvolvimento
Suporte
skype.softserv
(11) 3995 5200
(21) 3195 5200
(31) 3195 5200
(41) 3195 5200
(51) 3195 5200

Comercial
0800 727 2099
Ligação gratuita
São Paulo
Rio de Janeiro
Minas Gerais
Paraná
Rio Grande do Sul


Brasil
[Veja no ***@***.***,-43.271048,17z/data=!4m5!1m2!2m1!1ssoftserv!3m1!1s0x0000000000000000:0xa22666f336c06271?hl=pt-BR>    Sede
Av. Pastor Martin Luther King Jr. 126 Office 2000 Sala 226,
Shopping Nova América, Rio de janeiro - RJ, 20765-971   [http://softserv.com.br/email/assinatura/PENSE_IMPRIMIR.gif]
________________________________



________________________________
De: Simon Binder ***@***.***>
Enviado: terça-feira, 17 de agosto de 2021 15:11
Para: simolus3/moor ***@***.***>
Cc: Andre Fonseca ***@***.***>; Author ***@***.***>
Assunto: Re: [simolus3/moor] converting int into string on where clause (#1418)


You can cast the integer to a string like this:

t.nome + t.inscricao + t.clienteid.cast<String>()

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/simolus3/moor/discussions/1418#discussioncomment-1196722>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHLA7IQAUOCICVNU36ONBGDT5KQ5FANCNFSM5CKDUBMQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>.
